### DISCUSSION TITLE
Look for tutorials

### DISCUSSION THREAD

**Discussion by hangarr** [scikit-build/scikit-build#892] [Q&A]
I'm investigating _skbuild_ to build Python C++ extensions that use _opencv_.   We currently use _setuptools_ and _pybind11_, so other developers here are somewhat comfortable with that, but we are not expert as an organization doing this by any means.  It looks like _skbuild_ could be useful to us from these very simple examples:

> [**scikit-build-sample-projects**](https://github.com/scikit-build/scikit-build-sample-projects)
> `[](https://github.com/scikit-build/scikit-build-sample-projects)`
> 
> [**core-pybind11-hello**](https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/core-pybind11-hello)
> `[](https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/core-pybind11-hello)`
> 
> [**hello-pybind11**](https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/hello-pybind11)
> `[](https://github.com/scikit-build/scikit-build-sample-projects/tree/master/projects/hello-pybind11)`

We really need to find a good tutorial about using _skbuild_ with _pybind11_ to build C++ extensions that use OpenCV that we can follow to build our extensions.  Any pointers to tutorials like this would be greatly appreciated.  Thanks.

**Comment by henryiii** [comment]
I'd recommend https://github.com/pybind/scikit_build_example as a good example (it uses scikit-build-core, though an older version used classic scikit-build, if you'd rather that). I'm not aware of something specific with OpenCV, though. An example probably could be developed, though. Do you have any simple examples of something minimal with pybind11 + OpenCV that we could then try building?

**Comment by hangarr** [comment]
Thanks for the quick response. In between other tasks I happen to be building a very simple OpenCV example right now for other purposes.  I should have it in a couple of days and will post it then if that's OK.

**Reply by henryiii** [reply]
That would be great, I might be able to then adapt it to use scikit-build(-core?).

**Reply by hangarr** [reply]
Hi Henry,

i finally got the example pulled together.  It's in a public github repo:

**py11ocvex**
[https://github.com/hangarr/py11ocvex](https://github.com/hangarr/py11ocvex)

This probably doesn't conform to best Python conventions. It's a compromise with the structure of our code base.
